api/src/main/java/marquez/db/DatasetSymlinkDao.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2018-2022 contributors to the Marquez project
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package marquez.db;
+
+import java.time.Instant;
+import java.util.Optional;
+import java.util.UUID;
+import marquez.db.mappers.DatasetSymlinksRowMapper;
+import marquez.db.models.DatasetSymlinkRow;
+import org.jdbi.v3.sqlobject.config.RegisterRowMapper;
+import org.jdbi.v3.sqlobject.statement.SqlQuery;
+import org.jdbi.v3.sqlobject.statement.SqlUpdate;
+
+@RegisterRowMapper(DatasetSymlinksRowMapper.class)
+public interface DatasetSymlinkDao extends BaseDao {
+
+  default DatasetSymlinkRow upsertDatasetSymlinkRow(
+      UUID uuid, String name, UUID namespaceUuid, boolean isPrimary, Instant now) {
+    doUpsertDatasetSymlinkRow(uuid, name, namespaceUuid, isPrimary, now);
+    return findDatasetSymlinkByNamespaceUidAndName(namespaceUuid, name).orElseThrow();
+  }
+
+  @SqlQuery("SELECT * FROM dataset_symlinks WHERE namespace_uuid = :namespaceUuid and name = :name")
+  Optional<DatasetSymlinkRow> findDatasetSymlinkByNamespaceUidAndName(
+      UUID namespaceUuid, String name);
+
+  @SqlUpdate(
+      "INSERT INTO dataset_symlinks ("
```

Comment:
In [`V46__dataset_symlinks.sql`](https://github.com/MarquezProject/marquez/blob/feature/dataset-symlinks/api/src/main/resources/marquez/db/migration/V46__dataset_symlinks.sql) a `type` column is defined, but never set here. Do we need the column?

Reply:
This column is part of a spec to contain some description of an identifier for example: `table` may be a type for symlink names containing table name.

I forgot to make use of the column. Thanks for spotting this. 

I added some code to save a `type` and included it in unit test `marquez.db.OpenLineageDaoTest#testUpdateMarquezModelDatasetWithSymlinks` to make sure the type from OL event is written to Marquez DB.

Reply:
Thanks! ❤️